### ISSUE TITLE
MELCloud - Auth Code and Device Creation

### ISSUE THREAD

**Issue by sburke781** [sburke781/hubitat#30]
From @jasper on HE forum:

 "parent.getAuthCode returns null right after creation"

He also mentioned that in the setup notes the instruction to refresh handles this.  It would be nice to handle this more elegantly.  Need to analyse this in a little more detail to flesh out exactly what is happening.

**Comment by JasperVanLeeuwen** [comment]
some details: 
During the parent.refresh() method, the authcode is obtained and stored in the state, and in the same call the child drivers are created. 

The problem is that the state only gets persisted after execution. The child drivers try (upon creation) to obtain the authcode from the parent state, but that does not exist yet (resulting in authCode = null).

**Comment by sburke781** [comment]
You were spot on with your assessment of this @JasperVanLeeuwen , I experienced the same issue whilst developing my new set of drivers to replace these.  In the new "Unified Thermostat" drivers I resolved this by putting in a one-off schedule to run the device initialization 30 seconds after the authentication is completed.  Once I get some more of the notes and other stuff done in terms of releasing the new drivers, I would be interested in your feedback to see if I can close this issue off.  Thanks again for your assistance in getting these drivers up and going.

Simon
